src/main/hardwareWallet.ts

```diff
@@ -54,11 +57,13 @@ const connectHardwareWallet = async ({ lastIndexesByWallet }: TConnectHardwareWa
     }
   }
 
-  if (transporters.some(transport => transport.blockchain === 'neoLegacy')) {
-    transporters = transporters.filter(transport => transport.blockchain === 'neoLegacy')
-  }
+  if (newTransporters.length > 0) {
+    disconnectHardwareWallet()
 
-  if (!transporters.length) {
+    transporters = newTransporters.some(({ blockchain }) => blockchain === 'neoLegacy')
+      ? newTransporters.filter(({ blockchain }) => blockchain === 'neoLegacy')
+      : newTransporters
```

Comment:
Wha do you think ? This approach iterates through the array only once.
```suggestion
   const neoLegacyTransport = newTransporters.find(({ blockchain }) => blockchain === 'neoLegacy')
   transporters = neoLegacyTransport ? [neoLegacyTransport] : newTransporters

```